cloud/pkg/cloudhub/servers/httpserver/server.go

```diff
@@ -93,23 +94,59 @@ func EncodeCertPEM(cert *x509.Certificate) []byte {
 	return pem.EncodeToMemory(&block)
 }
 
-// edgeCoreClientCert will verify the token then create EdgeCoreCert and return it
+// edgeCoreClientCert will verify the certificate of EdgeCore or token then create EdgeCoreCert and return it
 func edgeCoreClientCert(w http.ResponseWriter, r *http.Request) {
+	if cert := r.TLS.PeerCertificates; len(cert) > 0 {
+		if err := verifyCert(cert[0]); err != nil {
+			klog.Errorf("failed to sign the certificate for edgenode: %s, failed to verify the certificate", r.Header.Get("NodeName"))
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte(err.Error()))
+		} else {
+			signEdgeCert(w, r)
+		}
+		return
+	}
+	if ok := verifyAuthorization(w, r); ok {
+		signEdgeCert(w, r)
+	} else {
+		klog.Errorf("failed to sign the certificate for edgenode: %s, invalid token", r.Header.Get("NodeName"))
+	}
```

Comment:
```suggestion
	if  !verifyAuthorization(w, r) {
		klog.Errorf("failed to sign the certificate for edgenode: %s, invalid token", r.Header.Get("NodeName"))
	} 
	signEdgeCert(w, r)
```

Reply:
If we do this, whether the token is right or not cloudcore will always sign the certificate for edgenode.

Reply:
ohh, if we change as my comments that we need return in `if`. So keep your original code is fine :)

Reply:
And seems that `ok` can be deleted.

Reply:
got it.